<a href="https://colab.research.google.com/github/cheungngo/MScGenomics/blob/master/5030/L4_BLAST.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# L4: BLAST

## Downloading the software

In [1]:
!wget "https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/ncbi-blast-2.12.0+-x64-linux.tar.gz" -O blast.tar.gz

--2022-02-22 16:31:43--  https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/LATEST/ncbi-blast-2.12.0+-x64-linux.tar.gz
Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 165.112.9.229, 165.112.9.230, 2607:f220:41e:250::11, ...
Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|165.112.9.229|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 247307631 (236M) [application/x-gzip]
Saving to: ‘blast.tar.gz’

blast.tar.gz        100%[===================>] 235.85M  34.9MB/s    in 7.4s    

2022-02-22 16:31:51 (32.0 MB/s) - ‘blast.tar.gz’ saved [247307631/247307631]



In [2]:
!gzip -d blast.tar.gz
!tar -xf blast.tar

In [6]:
!cp "/content/ncbi-blast-2.12.0+/bin" ../ -r # here we installed the program

### Exercise 1

Combine the following two genomes and then make a protein database using makeblastdb
command:


>e.coli_K-12_substr_MG1655_protein.faa

>r.denitrificans_protein.faa


Then use blast to search against the database for query sequences in seqs.fas. Report the number of
real homologous genes for each query sequence, based on the blast output

In [8]:
# copying the pre-downloaded database
!cp "/content/drive/MyDrive/Colab Notebooks/5030/L4" . -r

In [9]:
!cat "/content/L4/e.coli_K-12_substr_MG1655_protein.faa" "/content/L4/r.denitrificans_protein.faa" > newdb.faa

In [10]:
!makeblastdb -in newdb.faa -out db/mydb -dbtype prot -parse_seqids



Building a new DB, current time: 02/22/2022 16:34:55
New DB name:   /content/db/mydb
New DB title:  newdb.faa
Sequence type: Protein
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 8135 sequences in 0.283479 seconds.




In [11]:
!blastx -db db/mydb -query "/content/L4/seqs.fas" -out ex1.txt

BLASTX 2.12.0+


Reference: Stephen F. Altschul, Thomas L. Madden, Alejandro A.
Schaffer, Jinghui Zhang, Zheng Zhang, Webb Miller, and David J.
Lipman (1997), "Gapped BLAST and PSI-BLAST: a new generation of
protein database search programs", Nucleic Acids Res. 25:3389-3402.



Database: newdb.faa
           8,135 sequences; 2,598,789 total letters



Query= NC_000913.3:49823-50302 folA  Escherichia coli str. K-12 substr.
MG1655, complete genome

Length=480
                                                                      Score     E
Sequences producing significant alignments:                          (Bits)  Value

NP_414590.1 dihydrofolate reductase [Escherichia coli str. K-12 s...  333     5e-120
WP_011568103.1 dihydrofolate reductase [Roseobacter denitrificans]    95.1    4e-26 
NP_417794.1 bifunctional prepilin leader peptidase/ methylase [Es...  29.6    0.16  
NP_416077.4 Qin prophage; putative antitermination protein Q [Esc...  28.1    0.58  
WP_011570239.1 hemin import ATP-binding protein HmuV [Roseobacter...  27.3    0.94  
NP_415740.1 sensory histidine kinase in two-component regulatory ...  26.6    1.8   
WP_011569939.1 FAD-binding oxidoreductase [Roseobacter denitrific...  26.2    2.5   
NP_415497.1 cytochrome bd-II oxidase, subunit I [Escherichia coli...  26.2    2.6   
WP_085978967.1 acetyltransferase [Roseobacter denitrificans]          25.4    4.7   
NP_414763.1 RAYT REP element-mobilizing transposase; TnpA(REP) [E...  24.6    5.8   
WP_011568820.1 ornithine-acyl-ACP acyltransferase [Roseobacter de...  25.0    6.2   
NP_416967.1 N-succinyl-diaminopimelate deacylase [Escherichia col...  25.0    6.9   
WP_011568957.1 acetolactate synthase small subunit [Roseobacter d...  24.6    7.1   
WP_011568480.1 chromate transporter [Roseobacter denitrificans]       24.6    7.6   


### Exercise 2

In [ ]:
!wget "http://sgd-archive.yeastgenome.org/sequence/S288C_reference/orf_protein/orf_trans.fasta.gz"

--2022-02-16 10:22:19--  http://sgd-archive.yeastgenome.org/sequence/S288C_reference/orf_protein/orf_trans.fasta.gz
Resolving sgd-archive.yeastgenome.org (sgd-archive.yeastgenome.org)... 52.92.165.19
Connecting to sgd-archive.yeastgenome.org (sgd-archive.yeastgenome.org)|52.92.165.19|:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 2580699 (2.5M) [application/x-gzip]
Saving to: ‘orf_trans.fasta.gz’

orf_trans.fasta.gz  100%[===================>]   2.46M  2.62MB/s    in 0.9s    

2022-02-16 10:22:20 (2.62 MB/s) - ‘orf_trans.fasta.gz’ saved [2580699/2580699]



In [ ]:
!gzip -d orf_trans.fasta.gz

In [ ]:
!makeblastdb -in orf_trans.fasta -out db/ex2db -dbtype prot -parse_seqids



Building a new DB, current time: 02/16/2022 10:31:05
New DB name:   /content/db/ex2db
New DB title:  orf_trans.fasta
Sequence type: Protein
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 6034 sequences in 0.290888 seconds.




In [ ]:
!makeblastdb -in newdb.faa -out db/ex1db -dbtype prot -parse_seqids



Building a new DB, current time: 02/16/2022 10:31:35
New DB name:   /content/db/ex1db
New DB title:  newdb.faa
Sequence type: Protein
Keep MBits: T
Maximum file size: 1000000000B
Adding sequences from FASTA; added 8135 sequences in 0.292017 seconds.




In [ ]:
!blastp -db db/ex1db -query orf_trans.fasta -out ex2.txt

In [ ]:
!cp ex2.txt "/content/drive/MyDrive/Colab Notebooks/5030/L4_ex2.txt"